CRM/Pledge/BAO/PledgePayment.php

```diff
@@ -154,31 +154,11 @@ public static function create($params) {
    *   pledge payment id
    */
   public static function add($params) {
-    if (!empty($params['id'])) {
-      CRM_Utils_Hook::pre('edit', 'PledgePayment', $params['id'], $params);
-    }
-    else {
-      CRM_Utils_Hook::pre('create', 'PledgePayment', NULL, $params);
-    }
-
-    $payment = new CRM_Pledge_DAO_PledgePayment();
-    $payment->copyValues($params);
-
     // set currency for CRM-1496
-    if (!isset($payment->currency)) {
-      $payment->currency = CRM_Core_Config::singleton()->defaultCurrency;
+    if (empty($params['id']) && !isset($params['currency'])) {
+      $params['currency'] = CRM_Core_Config::singleton()->defaultCurrency;
```

Comment:
I think devs sometimes get confused that a function called "add" can also update. This is not the first time I've seen defaults get set even for update ops. I assume this was an oversight so I've fixed it by adding the `empty($params['id'])` condition.